embedded-value/src/main/java/com/iluwatar/embedded/value/DataSource.java

```diff
@@ -0,0 +1,217 @@
+import java.util.stream.Stream;
+
+import lombok.extern.slf4j.Slf4j;
+/*
+ * Communicates with database with the help of JDBC API
+ * 
+ * Inherits the SQL queries and methods from @link AbstractDataSource class
+ */
+
+@Slf4j
+public class DataSource implements DataSourceInterface {
+  private Connection conn;
+
+  /** 
+   * Statements are objects which are used to execute queries which will not be
+   * repeated.
+   */
+  private Statement createschema;
+  private Statement getschema;
+  private Statement deleteschema;
+  private Statement queryOrders;
+  
+  /*
+   * PreparedStatements are used to execute queries which will be repeated.
+   */
+  private PreparedStatement insertIntoOrders;
+  private PreparedStatement removeorder;
+  private PreparedStatement queyOrderByID;
+  
+  /**
+   * {@summary} 
+   * Establish connection to database.
+   */
+  public DataSource() {
+    try {
+      conn = DriverManager.getConnection(JDBC_URL);
+      LOGGER.info("Connected to H2 in-memory database: " + conn.getCatalog());
+    } catch (SQLException e) {
+      LOGGER.error(e.getLocalizedMessage(), e.getCause());
+    }
+  }
+
+  @Override
+  public boolean createSchema() {
+    try {
+      createschema = conn.createStatement();
+      createschema.execute(CREATE_SCHEMA);
+      insertIntoOrders = conn.prepareStatement(INSERT_ORDER, PreparedStatement.RETURN_GENERATED_KEYS);
+      getschema = conn.createStatement();
+      queryOrders = conn.createStatement();
+      removeorder = conn.prepareStatement(REMOVE_ORDER);
+      queyOrderByID = conn.prepareStatement(QUERY_ORDER);
+      deleteschema = conn.createStatement();
+      createschema.close();
+    } catch (SQLException e) {
+      LOGGER.error(e.getLocalizedMessage(), e.getCause());
+      return false;
+    }
+    return true;
+  }
+
+  @Override
+  public String getSchema() {
+    try {
+      var resultSet = getschema.executeQuery(GET_SCHEMA);
+      var sb = new StringBuilder();
+      while (resultSet.next()) {
+        sb.append("Col name: " + resultSet.getString(1) + ",  Col type: " + resultSet.getString(2) + "\n");
+      }
+      getschema.close();
+      return sb.toString();
+    } catch (Exception e) {
+      LOGGER.error("Error in retrieving schema: {}", e.getLocalizedMessage(), e.getCause());
+    }
+    return "Schema unavailable";
+  }
+  
+  @Override
+  public boolean insertOrder(Order order) {
+    try {
+      conn.setAutoCommit(false);
+      insertIntoOrders.setString(1, order.getItem());
+      insertIntoOrders.setString(2, order.getOrderedBy());
+      var address = order.getShippingAddress();
+      insertIntoOrders.setString(3, address.getCity());
+      insertIntoOrders.setString(4, address.getState());
+      insertIntoOrders.setString(5, address.getPincode());
+      var affectedRows = insertIntoOrders.executeUpdate();
+      if (affectedRows == 1) {
+        var rs = insertIntoOrders.getGeneratedKeys();
+        rs.last();
+        order.setId(rs.getInt(1));
```

Comment:
Why you are mutating the parameter instead of returning a new one?

Reply:
I'll return a new one then.